### ISSUE TITLE
assert at ICSharpCode.Decompiler.IL.Transforms.UsingTransform.TransformUsing

### ISSUE THREAD

**Issue by greenozon** [icsharpcode/ILSpy#1535]
ILSpy version 5.0.0.4804-preview2

Navigate to method
Microsoft.WindowsAzure.Storage.Blob.ParallelDownloadToFile.StartAsync

```
Error decompiling Microsoft.WindowsAzure.Storage.Blob.ParallelDownloadToFile.StartAsync
 ---> ICSharpCode.ILSpy.AssertionFailedException: 
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message, String detailMessage) in offset 380
   at ICSharpCode.ILSpy.ILSpyTraceListener.Fail(String message) in offset 9
   at System.Diagnostics.TraceInternal.Fail(String message) in offset 55
   at System.Diagnostics.Debug.Assert(Boolean condition) in offset 6
   at ICSharpCode.Decompiler.IL.ILVariable.set_Kind(VariableKind value) in offset 67
   at ICSharpCode.Decompiler.IL.Transforms.UsingTransform.TransformUsing(Block block, Int32 i) in offset 373
   at ICSharpCode.Decompiler.IL.Transforms.UsingTransform.ICSharpCode.Decompiler.IL.Transforms.IBlockTransform.Run(Block block, BlockTransformContext context) in offset 45
   at ICSharpCode.Decompiler.IL.Block.RunTransforms(IEnumerable`1 transforms, BlockTransformContext context) in offset 69
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.VisitBlock(ControlFlowNode cfgNode, BlockTransformContext context) in offset 146
   at ICSharpCode.Decompiler.IL.Transforms.BlockILTransform.Run(ILFunction function, ILTransformContext context) in offset 146
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 595
-- continuing with outer exception (ICSharpCode.Decompiler.DecompilerException) --
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DecompileBody(IMethod method, EntityDeclaration entityDecl, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 1083
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(IMethod method, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 249
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.DoDecompile(ITypeDefinition typeDef, DecompileRun decompileRun, ITypeResolveContext decompilationContext) in offset 783
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(IEnumerable`1 definitions) in offset 308
   at ICSharpCode.Decompiler.CSharp.CSharpDecompiler.Decompile(EntityHandle[] definitions) in offset 1
   at ICSharpCode.ILSpy.CSharpLanguage.DecompileType(ITypeDefinition type, ITextOutput output, DecompilationOptions options) in offset 79
   at ICSharpCode.ILSpy.TreeNodes.TypeTreeNode.Decompile(Language language, ITextOutput output, DecompilationOptions options) in offset 15
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.DecompileNodes(DecompilationContext context, ITextOutput textOutput) in offset 69
   at ICSharpCode.ILSpy.TextView.DecompilerTextView.<>c__DisplayClass34_0.<DecompileAsync>b__0() in offset 38

```

target: same as from #1534

**Comment by siegfriedpammer** [comment]
Sorry, I forgot to download the binary. Could you please upload it again? Thank you very much!

**Comment by greenozon** [comment]
https://www.sendspace.com/file/4oz623

**Comment by siegfriedpammer** [comment]
Thank you!

**Comment by dgrunwald** [comment]
This was fixed in beed6b5e24f990304 and 856cedc95e647fa744.
